dbt_meshify/storage/file_content_editors.py

```diff
@@ -0,0 +1,453 @@
+        """Add a group to a yml file"""
+        if groups_yml is None:
+            groups_yml = {}
+
+        groups = resources_yml_to_dict(groups_yml, NodeType.Group)
+        group_yml = groups.get(group.name) or {}
+
+        group_yml.update({"name": group.name})
+        owner = group_yml.get("owner", {})
+        owner.update(filter_empty_dict_items(group.owner.to_dict()))
+        group_yml["owner"] = owner
+
+        groups[group.name] = filter_empty_dict_items(group_yml)
+
+        groups_yml["groups"] = list(groups.values())
+        return groups_yml
+
+    @staticmethod
+    def add_access_to_model_yml(
+        model_name: str, access_type: AccessType, models_yml: Dict[str, Any]
+    ):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"access": access_type.value})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
+
+    @staticmethod
+    def add_group_to_model_yml(model_name: str, group: Group, models_yml: Dict[str, Any]):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models = resources_yml_to_dict(models_yml)
+        model_yml = models.get(model_name) or {"name": model_name, "columns": [], "config": {}}
+
+        model_yml.update({"group": group.name})
+        models[model_name] = process_model_yml(model_yml)
+
+        models_yml["models"] = list(models.values())
+        return models_yml
+
+    @staticmethod
+    def add_group_and_access_to_model_yml(
+        model_name: str, group: Group, access_type: AccessType, models_yml: Dict[str, Any]
+    ):
+        """Add group and access configuration to a model's YAMl properties."""
+        # parse the yml file into a dictionary with model names as keys
+        models_yml = DbtMeshFileEditor.add_access_to_model_yml(model_name, access_type, models_yml)
+        models_yml = DbtMeshFileEditor.add_group_to_model_yml(model_name, group, models_yml)
+        return models_yml
+
+    def get_source_yml_entry(
+        self, resource_name: str, full_yml: Dict[str, Any], source_name: str
+    ) -> Tuple[Dict[str, Any], Optional[Dict[str, Any]]]:
+        """
+        Remove a single source entry from a source defintion block, return source definition with single source entry and the remainder of the original
+        """
+        sources = resources_yml_to_dict(full_yml, NodeType.Source)
+        source_definition = sources.get(source_name)
+        tables = source_definition.get("tables", None)
+        table = list(filter(lambda x: x["name"] == resource_name, tables))
```

Comment:
Warning! A `None` type is not iterable.
```suggestion
        tables = source_definition.get("tables", [])
        table = list(filter(lambda x: x["name"] == resource_name, tables))
```